app/talk/comment-box.jsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import createReactClass from 'create-react-class';
+import ToggleChildren from './mixins/toggle-children';
+import Feedback from './mixins/feedback';
+import CommentImageSelector from './comment-image-selector';
+import getSubjectLocation from '../lib/getSubjectLocation';
+import Loading from '../components/loading-indicator';
+import SingleSubmitButton from '../components/single-submit-button';
+import alert from '../lib/alert';
+import {Markdown, MarkdownEditor, MarkdownHelp} from 'markdownz';
+import Suggester from './suggester';
+
+const ERROR_MESSAGES = {
+  BANNED: 'You are banned. Email contact@zooniverse.org for further information.',
+  EMAIL_NOT_CONFIRMED: 'Your account email address is not yet confirmed. Please check your email for confirmation instructions, or visit https://www.zooniverse.org/settings/email to request new confirmation email.'
+};
+
+function getErrorMessage(error, user) {
+  if (!error) return '';
+
+  if (
+    error.message?.match?.(/not allowed to create this Comment/i)
+    || error.message?.match?.(/not allowed to create this Discussion/i)
+    || error.message?.match?.(/You must confirm your account/i)
+  ) {
+    if (!user?.confirmed_at) {
+      return ERROR_MESSAGES.EMAIL_NOT_CONFIRMED;
+    }
+  } else if (
+    error.message?.match?.(/You are banned/i)
+  ) {
+    return ERROR_MESSAGES.BANNED;
+  }
+
+  return error.message || 'Unknown error';
+}
```

Comment:
Dev Notes 1/2: this is one of two additions/modifications to the CommentBox code.

In these blocks of code, we add a function for converting server-side error messages into something more detailed for the volunteer. Error message text is taken directly from #7021.

The rest of the changes is just a result of decaffeinating the CoffeeScript.